pandas/tests/frame/test_timeseries.py

```diff
@@ -323,6 +323,13 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        empty = Series(index=pd.DatetimeIndex([])).asfreq('H')
+        normal = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
+        self.assertEqual(empty.index.freq, normal.index.freq)
```

Comment:
@sinhrks should I add assert it for these types of indices in the same test or should I create separate tests for them?